src/Functions/if.cpp

```diff
@@ -42,6 +42,8 @@ using namespace GatherUtils;
 /** Selection function by condition: if(cond, then, else).
   * cond - UInt8
   * then, else - numeric types for which there is a general type, or dates, datetimes, or strings, or arrays of these types.
+  * For better performance, try to use branch free code for numeric types except floating point types because of inf or nan, besides the trick will not improve much on floating point types. eg.
```

Comment:
Cosmetic: Could you please put in a newline between l. 44 and l. 55. Also, the statement on l. 45 is kind of ambiguous. Maybe you meant: `For better performance, try to use branch-free evaluation for integral types (i.e. cond ? a : b --> !!cond * a + !cond * b). This doesn't apply to floating-point types because of Inf and NaN.`